pkg/feature/api/flag_trigger.go

```
@@ -17,62 +17,987 @@ package api
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return resp, nil
+	}
+	if trigger.GetAction() == featureproto.FlagTrigger_Action_ON {
+		err := s.enableFeature(ctx, trigger.GetFeatureId(), trigger.GetEnvironmentNamespace(), localizer)
+		if err != nil {
+			return resp, nil
+		}
+	} else if trigger.GetAction() == featureproto.FlagTrigger_Action_OFF {
+		err := s.disableFeature(ctx, trigger.GetFeatureId(), trigger.GetEnvironmentNamespace(), localizer)
+		if err != nil {
+			return resp, nil
+		}
+	} else {
+		s.logger.Error(
+			"Invalid action",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return resp, nil
+	}
+	err = s.updateTriggerUsageInfo(ctx, trigger)
+	if err != nil {
+		return resp, nil
+	}
+	return resp, nil
+}
+
+func (s *FeatureService) authSecret(
+	ctx context.Context,
+	secret string,
+) (*domain.FlagTriggerSecret, error) {
+	decoded, err := base64.RawURLEncoding.DecodeString(secret)
+	if err != nil {
+		s.logger.Error(
+			"Failed to decode encrypted secret",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return nil, err
+	}
+	decrypted, err := s.triggerCryptoUtil.Decrypt(ctx, decoded)
+	if err != nil {
+		s.logger.Error(
+			"Failed to decrypt encrypted secret",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return nil, err
+	}
+	triggerSecret, err := domain.UnmarshalFlagTriggerSecret(decrypted)
+	if err != nil {
+		s.logger.Error(
+			"Failed to unmarshal trigger url secret",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return nil, err
+	}
+	return triggerSecret, nil
+}
+
+func (s *FeatureService) updateTriggerUsageInfo(ctx context.Context, flagTrigger *domain.FlagTrigger) error {
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).
+				AddFields(zap.Error(err))...,
+		)
+		return nil
```

Please return the error.